controllers/util/storageconsumer.go

```diff
@@ -155,3 +156,22 @@ func (wrapper storageConsumerResourceConfigMapWrapper) SetCephFsClientProfileNam
 func (wrapper storageConsumerResourceConfigMapWrapper) SetNfsClientProfileName(name string) {
 	wrapper.Data[nfsClientProfileKey] = name
 }
+
+func GenerateStorageConsumerResourceDefaults(storageConsumerName, clientProfileName string) map[string]string {
+	defaults := &corev1.ConfigMap{}
+	defaults.Data = map[string]string{}
+	defaultConfig := WrapStorageConsumerResourceConfigMap(defaults)
+	defaultConfig.SetRbdRadosNamespaceName(storageConsumerName)
+	defaultConfig.SetSubVolumeGroupName(storageConsumerName)
+	defaultConfig.SetSubVolumeGroupRadosNamespaceName(storageConsumerName)
+	defaultConfig.SetRbdClientProfileName(clientProfileName)
+	defaultConfig.SetCephFsClientProfileName(clientProfileName)
+	defaultConfig.SetNfsClientProfileName(clientProfileName)
+	defaultConfig.SetCsiRbdNodeSecretName(fmt.Sprintf("rbd-node-%s", clientProfileName))
+	defaultConfig.SetCsiRbdProvisionerSecretName(fmt.Sprintf("rbd-provisioner-%s", clientProfileName))
+	defaultConfig.SetCsiCephFsNodeSecretName(fmt.Sprintf("cephfs-node-%s", clientProfileName))
+	defaultConfig.SetCsiCephFsProvisionerSecretName(fmt.Sprintf("cephfs-provisioner-%s", clientProfileName))
+	defaultConfig.SetCsiNfsNodeSecretName(fmt.Sprintf("nfs-node-%s", clientProfileName))
+	defaultConfig.SetCsiNfsProvisionerSecretName(fmt.Sprintf("nfs-provisioner-%s", clientProfileName))
+	return defaults.Data
```

Comment:
There is no point in having the wrapper work on top of the config map, because it requires that you need to have a configmap for it to work. I prefer if we can change the wrapper to work over the map itself